File: app/src/at-dinhtruong/java/asiantech/internship/summer/restapi/ImageAdapter.java
package asiantech.internship.summer.restapi;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import java.util.List;

import asiantech.internship.summer.R;
import asiantech.internship.summer.models.Image;

public class ImageAdapter extends RecyclerView.Adapter<ImageAdapter.ImageViewHolder> {
    private List<Image> mImages;

    ImageAdapter(List<Image> images) {
        mImages = images;
    }

    @NonNull
    @Override
    public ImageViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(viewGroup.getContext());
        View itView = layoutInflater.inflate(R.layout.restapi_item, viewGroup, false);
        return new ImageAdapter.ImageViewHolder(itView);
    }

    @Override
    public void onBindViewHolder(@NonNull ImageViewHolder holder, int position) {
        holder.onBind(mImages.get(position));
    }

    @Override
    public int getItemCount() {
        return mImages.size();
    }

    class ImageViewHolder extends RecyclerView.ViewHolder {
        private ImageView mImgItem;
        private TextView mTvItem;

        ImageViewHolder(@NonNull View itView) {
            super(itView);
            mImgItem = itView.findViewById(R.id.imgItem);
            mTvItem = itView.findViewById(R.id.tvItem);
        }

        private void onBind(Image image) {
            if (image != null) {
                mTvItem.setText(image.getImageId());
                Glide.with(itemView.getContext()).load(image.getUrl()).into(mImgItem);
            }
        }
    }
}

File: app/src/at-dinhtruong/java/asiantech/internship/summer/kotlin/retrofit/ImageAdapter.kt
package asiantech.internship.summer.kotlin.retrofit

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import asiantech.internship.summer.R
import asiantech.internship.summer.kotlin.model.Image
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.`at-dinhtruong`.restapi_item.view.*


class ImageAdapter(private val listImage: ArrayList<Image>) : RecyclerView.Adapter<ImageAdapter.ImageViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup,
                                    viewType: Int): ImageAdapter.ImageViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.restapi_item, parent, false)
        return ImageViewHolder(view)
    }

    override fun onBindViewHolder(holder: ImageViewHolder, position: Int) {
        holder.onBind()
    }

    override fun getItemCount() = listImage.size

    inner class ImageViewHolder(val view: View) : RecyclerView.ViewHolder(view) {
        fun onBind() {
            val image = listImage.get(index = adapterPosition)
            if (!image.image_id.isEmpty()) {
                Glide.with(itemView.context).load(image.url).into(itemView.imgItem)
                view.tvItem.text = image.image_id
            }
        }
    }
}
